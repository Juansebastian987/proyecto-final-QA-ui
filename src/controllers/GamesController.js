import axios from 'axios';

const urlBase = process.env.API;

export const GamesController = {

    register(game) {
        return axios({
            method: 'POST',
            baseURL: urlBase,
            url: `games`,
            data: game,
        })
    },
    list() {
        return axios({
            method: 'GET',
            baseURL: urlBase,
            url: 'games'
        });
    },
    delete(name) {
        return axios({
            method: 'DELETE',
            baseURL: urlBase,
            url: `games/${name}`,
        });
    },
    getGame(name) {
        return axios({
            method: 'GET',
            baseURL: urlBase,
            url: `games/${name}`,
        });
    },
    updateGame(name) {
        return axios({
            method: 'PUT',
            baseURL: urlBase,
            url: `games/${name}`,
        });
    }
}