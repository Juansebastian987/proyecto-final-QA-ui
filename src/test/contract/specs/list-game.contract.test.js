const { provider } = require("../config/init-pact");
const { Matchers } = require("@pact-foundation/pact");
import { GamesController } from  '../../../controllers'

describe("Given an game service", () => {
  describe("When a request to list games is made", () => {
    beforeAll(async () => {
      await provider.setup();
      await provider.addInteraction({
        state: "There are games",
        uponReceiving: "a request to get all games",
        withRequest: {
          method: "GET",
          path: "/games",
        },
        willRespondWith: {
          status: 200,
          body: Matchers.eachLike(
            {
              id: Matchers.like("1"),
              name: Matchers.like("Borderlands 3"),
              developer: Matchers.like("GearBox Software"),
              year: Matchers.like(2019),
              type: Matchers.like("Shooter"),
              console: Matchers.like("pc"),
            },
            { min: 1 }
          ),
        },
      });
    });

    it("Should return the correct data", async () => {
      const response = await GamesController.list();
      expect(response.data).toMatchSnapshot();

      await provider.verify()

    });

    afterAll(async () => {
      await provider.finalize();
    });
  });
});
