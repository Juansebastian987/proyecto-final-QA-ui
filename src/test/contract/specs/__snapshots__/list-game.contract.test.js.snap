// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an game service When a request to list games is made Should return the correct data 1`] = `
Array [
  Object {
    "console": "pc",
    "developer": "GearBox Software",
    "id": "1",
    "name": "Borderlands 3",
    "type": "Shooter",
    "year": 2019,
  },
]
`;
